
<launch>
	
  <arg name="master_name" default="phantom_master"/>
  <arg name="slave_name" default="grips"/>
  <arg name="ref_frame" default="world"/>
  <arg name="mode_name" default="a"/>    <!-- Mode: a, b , c -->
  <arg name="rpy_offset" default="[0, 0, 0]"/>
  <arg name="rviz" default="true"/>
  
  <!-- Haptic device interface -->
  <node name="haptic_device_interface" pkg="rate_position_controller" type="haptic_device_interface.py" output="screen"> 
    <param name="device_name" type="string" value="$(arg master_name)"/>
    <param name="read_ip" type="string" value="192.168.0.108"/>
    <param name="write_ip" type="string" value="192.168.0.5"/>
    <param name="read_port" type="int" value="5053"/>
    <param name="write_port" type="int" value="5054"/>
  </node>
  
  <node name="send_udp_force" pkg="rate_position_controller" type="send_udp_force.py" output="screen"> 
    <param name="device_name" type="string" value="$(arg master_name)"/>
    <param name="write_ip" type="string" value="192.168.0.5"/>
    <param name="write_port" type="int" value="5054"/>
    <param name="publish_rate" type="double" value="1000.0"/>
  </node>
  
<!-- Rate position controller -->
  <node name="rate_pos_controller" pkg="rate_position_controller" type="rate_pos_controller.py" output="screen">
    <param name="publish_rate" type="double" value="1000"/>
    <param name="slave_name" type="string" value="$(arg slave_name)"/>
    <param name="master_name" type="string" value="$(arg master_name)"/>
    <param name="reference_frame" type="string" value="$(arg ref_frame)"/>
    <param name="mode_name" type="string" value="$(arg mode_name)"/>
    <rosparam param="rpy_offset" subst_value="True">$(arg rpy_offset)</rosparam>
    <rosparam command="load" file="$(find rate_position_controller)/config/phantom_kraft.yaml" />
  </node>
  
  <!-- Force handler node -->
<!--
  <node name="force_handler" pkg="rate_position_controller" type="force_handler.py" output="screen"> 
    <param name="publish_rate" type="double" value="1000"/>
    <param name="slave_name" type="string" value="$(arg slave_name)"/>
    <param name="master_name" type="string" value="$(arg master_name)"/>
    <param name="reference_frame" type="string" value="$(arg ref_frame)"/>
    <rosparam command="load" file="$(find rate_position_controller)/config/phantom_justin.yaml" />
  </node>
-->

  <!-- Strategies node -->
  <node name="strategies" pkg="rate_position_controller" type="strategies.py" output="screen"> 
    <param name="slave_name" type="string" value="$(arg slave_name)"/>
    <param name="master_name" type="string" value="$(arg master_name)"/>
    <param name="mode_name" type="string" value="$(arg mode_name)"/>
    <rosparam command="load" file="$(find rate_position_controller)/config/phantom_kraft.yaml" />
  </node>
  
  <!-- Show in Rviz -->
  <group if="$(arg rviz)">
    <node name="rviz" pkg="rviz" type="rviz" output="screen" args="-d $(find grips_teleop)/config/cartesian_controller.rviz"/>
  </group>
  
</launch>
